route/build/src/tracer.f90

```diff
@@ -0,0 +1,364 @@
+
+CONTAINS
+
+ ! *********************************************************************
+ ! subroutine: perform diffusive wave routing for one segment
+ ! *********************************************************************
+ SUBROUTINE constituent_rch(iens, segIndex,   & ! input: index of runoff ensemble to be processed
+                            idxRoute,         & ! input: routing method index
+                            ixDesire,         & ! input: reachID to be checked by on-screen pringing
+                            NETOPO_in,        & ! input: reach topology data structure
+                            RPARAM_in,        & ! input: reach parameter data structure
+                            RCHSTA_out,       & ! inout: reach state data structure
+                            RCHFLX_out,       & ! inout: reach flux data structure
+                            ierr, message)      ! output: error control
+
+ implicit none
+ ! Argument variables
+ integer(i4b),     intent(in)                 :: iens              ! runoff ensemble to be routed
+ integer(i4b),     intent(in)                 :: segIndex          ! segment where routing is performed
+ integer(i4b),     intent(in)                 :: idxRoute          ! routing method index
+ integer(i4b),     intent(in)                 :: ixDesire          ! index of the reach for verbose output
+ type(RCHTOPO),    intent(in),    allocatable :: NETOPO_in(:)      ! River Network topology
+ type(RCHPRP),     intent(inout), allocatable :: RPARAM_in(:)      ! River reach parameter
+ type(STRSTA),     intent(inout)              :: RCHSTA_out(:,:)   ! reach state data
+ type(STRFLX),     intent(inout)              :: RCHFLX_out(:,:)   ! Reach fluxes (ensembles, space [reaches]) for decomposed domains
+ integer(i4b),     intent(out)                :: ierr              ! error code
+ character(*),     intent(out)                :: message           ! error message
+ ! Local variables
+ logical(lgt)                                 :: verbose           ! check details of variables
+ logical(lgt)                                 :: isHW              ! headwater basin?
+ integer(i4b)                                 :: nUps              ! number of upstream segment
+ integer(i4b)                                 :: iUps              ! upstream reach index
+ integer(i4b)                                 :: iRch_ups          ! index of upstream reach in NETOPO
+ type(SUBRCH)                                 :: subrch_state      ! curent states at sub-reaches
+ real(dp)                                     :: Clat              ! lateral flow into channel [m3/s]
+ real(dp)                                     :: Cupstream         ! total discharge at top of the reach [m3/s]
+ character(len=strLen)                        :: cmessage          ! error message from subroutine
+
+ ierr=0; message='constituent_rch/'
+
+ verbose = .false.
+ if(NETOPO_in(segIndex)%REACHIX == ixDesire)then
```

Comment:
Why is verbose set to true in this case?

Reply:
I will comment on this in another PR.  

better idea would be using a function like this (may not need comment)
verbose = print_reach_states(reach_id)